CustomAugments.cpp

```diff
@@ -487,6 +487,24 @@ HOOK_METHOD_PRIORITY(ShipObject, GetAugmentationValue, 1000, (const std::string&
     return ret;
 }
 
+HOOK_METHOD_PRIORITY(ShipObject, GetAugmentationValue, 500, (const std::string& name) -> float)
```

Comment:
The GetAugmentationValue hook that overwrites the vanilla function has a priority of 1000. From what I can tell, since this has a priority of 500, any modifications it makes to the augment values will come after that base function. Vert says the reason is probably more sophisticated than that though, so you'll probably wanna hear from them.

Reply:
No vert has given me a more detailed explanation. If you have this code:
```c++
HOOK_METHOD_PRIORITY(Class, Method, 1000, () -> void)
{
  print("Priority 1000 Before. \n");
  super();
  print("Priority 1000 After. \n");
}
HOOK_METHOD_PRIORITY(Class, Method, 500, () -> void)
{
  print("Priority 500 Before. \n");
  super();
  print("Priority 500 After. \n");
}```
you get this output:
```
Priority 500 Before.
Priority 1000 Before.
Priority 1000 After.
Priority 500 After.```
So the old GetAugmentationValue runs its code, this new GetAugmentationValue gets it's return value and then it runs its code.

Reply:
So vert has given me a more detailed explanation. If you have this code, for example:
```c++
HOOK_METHOD_PRIORITY(Class, Method, 1000, () -> void)
{
  print("Priority 1000 Before. \n");
  super();
  print("Priority 1000 After. \n");
}
HOOK_METHOD_PRIORITY(Class, Method, 500, () -> void)
{
  print("Priority 500 Before. \n");
  super();
  print("Priority 500 After. \n");
}
```
you would get this output:
```
Priority 500 Before.
Priority 1000 Before.
Priority 1000 After.
Priority 500 After.
```
The way this is set up, the old GetAugmentationValue hook runs its code, then this new GetAugmentationValue hook gets its return value and runs *its* code.

Reply:
Yes but if you don't supply a priority then you're just in the default ordering.

We try to only supply a priority if it's 100% important that we be the *first* or *last* hook in the chain of hooks, we try not to manually specify priorities for hooks because that would get really painful to maintain.

Reply:
Fixed